File: VB/WebSite/App_Code/XpoRole.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports System.Collections.Generic

Public Class XpoRole
	Inherits XPObject

	Public Sub New()
		MyBase.New()
	End Sub

	Public Sub New(ByVal session As Session)
		MyBase.New(session)
	End Sub

	Public Overrides Sub AfterConstruction()
		MyBase.AfterConstruction()
	End Sub

	Private _ApplicationName As String
	<Size(255), Indexed("RoleName", Unique := True)> _
	Public Property ApplicationName() As String
		Get
			Return _ApplicationName
		End Get
		Set(ByVal value As String)
			SetPropertyValue("ApplicationName", _ApplicationName, value)
		End Set
	End Property

	Private _RoleName As String
	<Size(255), Indexed("ApplicationName", Unique := True)> _
	Public Property RoleName() As String
		Get
			Return _RoleName
		End Get
		Set(ByVal value As String)
			SetPropertyValue("RoleName", _RoleName, value)
		End Set
	End Property

	Private _Description As String
	<Size(255)> _
	Public Property Description() As String
		Get
			Return _Description
		End Get
		Set(ByVal value As String)
			SetPropertyValue("Description", _Description, value)
		End Set
	End Property

	Private _CreationDate As DateTime
	Public Property CreationDate() As DateTime
		Get
			Return _CreationDate
		End Get
		Set(ByVal value As DateTime)
			SetPropertyValue("CreationDate", _CreationDate, value)
		End Set
	End Property

	<Association("XpoRoles-XpoUsers")> _
	Public ReadOnly Property Users() As XPCollection(Of XpoUser)
		Get
			Return GetCollection(Of XpoUser)("Users")
		End Get
	End Property
End Class

File: CS/WebSite/App_Code/XpoRole.cs
using System;
using System.Linq;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using System.Collections.Generic;

public class XpoRole : XPObject {

    public XpoRole()
        : base() { }

    public XpoRole(Session session)
        : base(session) { }

    public override void AfterConstruction() {
        base.AfterConstruction();
    }

    private string _ApplicationName;
    [Size(255)]
    [Indexed("RoleName", Unique = true)]
    public string ApplicationName {
        get {
            return _ApplicationName;
        }
        set {
            SetPropertyValue("ApplicationName", ref _ApplicationName, value);
        }
    }

    private string _RoleName;
    [Size(255)]
    [Indexed("ApplicationName", Unique = true)]
    public string RoleName {
        get {
            return _RoleName;
        }
        set {
            SetPropertyValue("RoleName", ref _RoleName, value);
        }
    }

    private string _Description;
    [Size(255)]
    public string Description {
        get {
            return _Description;
        }
        set {
            SetPropertyValue("Description", ref _Description, value);
        }
    }

    private DateTime _CreationDate;
    public DateTime CreationDate {
        get {
            return _CreationDate;
        }
        set {
            SetPropertyValue("CreationDate", ref _CreationDate, value);
        }
    }

    [Association("XpoRoles-XpoUsers")]
    public XPCollection<XpoUser> Users {
        get {
            return GetCollection<XpoUser>("Users");
        }
    }
}

